e2e/app/page/notebook-page.ts

```diff
@@ -130,45 +137,41 @@ export default class NotebookPage extends NotebookFrame {
     return this.downloadAs(Xpath.downloadMarkdownButton);
   }
 
-  /**
-   * Wait for notebook kernel becomes ready (idle).
-   */
-  async waitForKernelIdle(timeOut?: number): Promise<void> {
+  async isIdle(timeout?: number): Promise<boolean> {
     const frame = await this.getIFrame();
     const idleIconSelector = `${CssSelector.kernelIcon}.kernel_idle_icon`;
     const notificationSelector = '#notification_kernel';
-    const isIdle = async (timeout): Promise<boolean> => {
-      return Promise.all([
-        frame.waitForSelector(idleIconSelector, { visible: true, timeout }),
-        frame.waitForSelector(notificationSelector, { hidden: true, timeout })
-      ])
-        .then(() => {
-          return true;
-        })
-        .catch(() => {
-          return false;
-        });
-    };
+    return Promise.all([
+      frame.waitForSelector(idleIconSelector, { visible: true, timeout }),
+      frame.waitForSelector(notificationSelector, { hidden: true, timeout })
+    ])
+      .then(() => {
+        return true;
+      })
+      .catch(() => {
+        return false;
+      });
+  }
+
+  /**
+   * Wait for notebook kernel becomes ready (idle).
+   */
+  async waitForKernelIdle(timeOut = 300000, sleepInterval = 10000): Promise<boolean> {
     // Check kernel status twice with a pause between two checks because kernel status can suddenly become not ready.
     let ready = false;
     const startTime = Date.now();
-    while (Date.now() - startTime <= timeOut) {
-      const idle = await isIdle(30000);
+    while (Date.now() - startTime < timeOut) {
+      const idle = await this.isIdle(2000);
       if (ready && idle) {
-        break;
+        return true;
       }
       ready = idle;
-      await this.page.waitForTimeout(5000);
+      if (Date.now() - startTime + sleepInterval < timeOut) {
```

Comment:
Want to avoid unnecessary sleep if wait time will run out during the sleep. I don't feel strongly about it, if it makes code messier, I can remove it.

Reply:
my preference would be to only use the condition in the while loop. it's slightly cleaner and the timeouts are arbitrary anyways

Reply:
I agree. the check is removed.